spock-specs/src/test/resources/org/spockframework/smoke/Interceptors__non_method_interceptors_are_called_even_without_the_respective_method.txt

```diff
@@ -0,0 +1,66 @@
+shared initializer start (SubSpec)
+  shared initializer start (SuperSpec)
+  shared initializer end (SuperSpec)
+shared initializer end (SubSpec)
+specification start (SubSpec)
+  setup spec start (SubSpec)
+    setup spec start (SuperSpec)
+    setup spec end (SuperSpec)
+  setup spec end (SubSpec)
+  initializer start (SubSpec.foo)
+    initializer start (SubSpec.foo)
+    initializer end (SubSpec.foo)
+  initializer end (SubSpec.foo)
+  feature start (SubSpec.foo)
+    iteration start (SubSpec.foo[#0])
+      setup start (SubSpec.foo[#0])
+        setup start (SubSpec.foo[#0])
+        setup end (SubSpec.foo[#0])
+      setup end (SubSpec.foo[#0])
```

Comment:
🤔 From a reporting POV it isn't ideal that you get the same spec name twice here, as one of the invoked methods is from the SuperSpec, although it is correct that the the feature belongs to the `SubSpec`. The same applied to the other methods.

Reply:
I agree, but that is nothing I change here.
I just add a test that verifies the current state, like I also do not fix the simple feature lifecycle here, but in a separate upstack PR.

So this simply is, what you get from `IMethodInvocation#spec#name` currently.

Maybe it should generally be reviewed whether the delivered information makes sense in all cases.

But actually I don't see how "one of the invoked methods is from the SuperSpec" here.
The super spec does not have any setup method.
It is just the super specs "wrapper around all setup methods" that is called for the setup phase of the sub spec.

If you have `setup` method in both sub spec and super spec like in the other feature, it looks like this:
```plain
setup start (SubSpec.subFeature1[#0])
  setup start (SubSpec.subFeature1[#0])
    setup method start (SuperSpec.subFeature1[#0].setup())
      fixture method start (SuperSpec.subFeature1[#0].setup())
      fixture method end (SuperSpec.subFeature1[#0].setup())
    setup method end (SuperSpec.subFeature1[#0].setup())
  setup end (SubSpec.subFeature1[#0])
  setup method start (SubSpec.subFeature1[#0].setup())
    fixture method start (SubSpec.subFeature1[#0].setup())
    fixture method end (SubSpec.subFeature1[#0].setup())
  setup method end (SubSpec.subFeature1[#0].setup())
setup end (SubSpec.subFeature1[#0])
```
But it might also be questionable whether this is correct.
Should it really say the spec is super, which matches the actual method called,
but the feature is the one from the sub spec which does not match the feature?

Maybe it would make more sense if spec here is indeed the sub spec, matching the feature you get, and if you want to know from where the method is from, you use `method.parent`?

I'm not sure how it best should be to be honest. For "shared initializer" for example, you get sub and super, not twice sub like here.

Reply:
Or maybe it makes sense that for the actual "...method" one the `IMethodInvocation.spec` gives the one matching that method as it is the main target in view currently, and if you want the features spec, you use `feature.parent`.

Reply:
Actually, in my other upstack PR I add tests for whether the fields are set at all or not,
but they don't verify to what they are set and whether the spec is consistent with the feature and so on. I guess another PR could introduce such checks later on and fix some of this if it is not like intended.

Reply:
I at least improved the reporting in the method-interceptor cases above now, so that the features are not attributed to the wrong specification.

Reply:
I was not saying that we should change anything on the IMethodInvocation side, but it would be possible to change

```groovy
specInfo.specsBottomToTop*.addSetupInterceptor {
        proceed(it, 'setup', "$it.spec.name.$it.feature.name[#$it.iteration.iterationIndex]")
      }
```
to 
```groovy
specInfo.specsBottomToTop.each { spec -> spec.addSetupInterceptor {
        proceed(it, 'setup', "$spec.name.$it.feature.name[#$it.iteration.iterationIndex]")
      }
}
```

Reply:
> I just add a test that verifies the current state, like I also do not fix the simple feature lifecycle here, but in a separate upstack PR.

So, the intention is to visualize which interceptors are called in what order. Given that the information in `IMethodInvocation` is no always sufficient for this, I think it makes sense to change the interceptor registrations to include the spec, like I've shown above.

Reply:
Oh, now I got it, yeah, makes absolute sense. Fixed now.